crates/biome_js_analyze/src/lint/nursery/use_default_case.rs

```diff
@@ -0,0 +1,195 @@
+        version: "next",
+        name: "useDefaultCase",
+        sources: &[RuleSource::Eslint("default-case")],
+        recommended: false,
+    }
+}
+
+/// Options for the rule `useDefaultCase`.
+#[derive(Clone, Debug, Default, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct UseDefaultCaseOptions {
+    /// Regular expression that filters the comment pattern to disable the rule.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub comment_pattern: Option<String>,
+}
+
+impl Rule for UseDefaultCase {
+    type Query = Ast<JsSwitchStatement>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = Box<UseDefaultCaseOptions>;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let comment_pattern = get_comment_pattern(ctx);
+
+        let has_case_clauses = node.cases().into_iter().len() > 0;
+        let is_missing_default_case = node
+            .cases()
+            .into_iter()
+            .any(|clause| clause.as_js_default_clause().is_some())
+            .not();
+
+        let curly_token = node.r_curly_token().ok();
+        let trivia_in_switch_block = curly_token.map(|c| c.leading_trivia());
+
+        let has_disable_comment_in_switch_block =
+            has_disable_comment(trivia_in_switch_block, &comment_pattern);
+
+        let has_disable_comment_in_switch_expression = node
+            .discriminant()
+            .ok()
+            .into_iter()
+            .filter(|token| token.as_js_identifier_expression().is_some())
+            .any(|token| {
+                let identifier_expression = token.as_js_identifier_expression().unwrap();
+
+                if let Ok(switch_identifier) = identifier_expression.name() {
+                    if let Ok(token) = switch_identifier.value_token() {
+                        if has_disable_comment(Some(token.trailing_trivia()), &comment_pattern) {
+                            return true;
+                        }
+                    }
+                }
+
+                false
+            });
+
+        let is_valid = is_missing_default_case
+            && has_case_clauses
+            && !has_disable_comment_in_switch_block
+            && !has_disable_comment_in_switch_expression;
+
+        is_valid.then_some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        let comment_pattern = get_comment_pattern(ctx);
+
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! {
+                    "Expected a default case."
+                },
+            )
+            .note(markup! {
+                "The lack of a default clause can result in unexpected behaviors in your code."
```

Comment:
This message seems confusing to me. I suggest something like `The lack of a default clause can be a possible omission.`

Reply:
done